src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/vmware/VMWareProvider.java

```diff
@@ -164,22 +200,34 @@ private VirtualMachineCloneSpec createVirtualMachineCloneSpec(int instanceIndexS
     }
 
     /**
-     * Generate a new VirtualMachineCloneSpec with parameters by default and based on the provided VM to clone
+     * Create a new VirtualMachineCloneSpec with parameters by default
      *
-     * @param vmToClone the source VM to rely on
      * @param instance  the current instance to rely on
      * @return a new customized VirtualMachineCloneSpec
      */
-    private VirtualMachineCloneSpec generateDefaultVirtualMachineCloneSpec(VirtualMachine vmToClone,
-            Instance instance) {
+    private VirtualMachineCloneSpec generateDefaultVirtualMachineCloneSpec(Instance instance) {
         VirtualMachineCloneSpec vmCloneSpecs = new VirtualMachineCloneSpec();
-        vmCloneSpecs.setLocation(vmWareProviderVirtualMachineUtil.getVirtualMachineRelocateSpec(vmToClone));
         vmCloneSpecs.setPowerOn(true);
         vmCloneSpecs.setTemplate(false);
         vmCloneSpecs.setConfig(getVirtualMachineConfigSpec(instance));
         return vmCloneSpecs;
     }
 
+    private VirtualMachineRelocateSpec generateCustomRelocateSpecs(VirtualMachine vmToClone,
+            ResourcePool destinationPool, HostSystem destinationHost, Datastore destinationDatastore)
```

Comment:
can you change destinationHost and destinationDatastore to be Optional method parameters (destinationHostOptional,destinationDatastoreOptional ) and then do destinationHostOptional.ifPresent(destinationHost -> vmRelocateSpecs.setHost(destinationHost.getMOR()))